app-experimental.tex

```diff
@@ -1311,6 +1315,69 @@ \subsection{Points-to-PCC} % <<<
 
 Any capability authorizing capability load may be made into a sealed indirect
 entry capability, for which we propose reserving the \cotype{} $2^{64} - 4$.%
+A new, two-operand instruction is required for sealing an isentry, which we call
+\insnnoref{CSealIndEntry}. This instruction is comparable in its behavior to the
+\insnref{CSealEntry}.
+
+However, we do realize that \insnnoref{CInvokeInd} is complex instruction and it imposes to have
+micro-operations in microarchitectures. Thus we propose to have the two operations
+of the complex \insnnoref{CInvokeInd} to be split into a linked pair of instructions:
+an instruction for load of the code capability and an instruction for jumping to the
+code capability and unsealing the data capability.
+This property of having linked will retain the atomicity of the both IDC and PCC
+being written with valid capabilities or none of them being written.
+
+\subsubsection{Definition of linked pair of instructions}
+
+In this extension, we propose to split the cisentryj (CHERI indirect sealed entry jump) instruction into two sub instructions: Lil (load isentry linked) and cjaurl (CHERI jump and unseal into register linked).
+In the following paragraphs, we will explain what these instructions do and how they enforce the security model we are designing.
+
+The lil instruction loads a capability through an isentry capability and clears its tag.
```

Comment:
Boom, we were discussing, and I reckon we can make this a lot nicer. Instead of having this loaded code cap being "sort of untagged but sort of not", we can just have a dedicated otype that says "only usable if it's being used in a cjaurl as the code cap with a valid reservation". When someone sets up one of these objects, they seal the code cap with that otype, and then because it's sealed it can't be used by anyone for anything else. That will also prevent the revoker from missing it, and presumably all sorts of other problems that we hadn't thought of yet due to these partially untagged values

Reply:
Hmm, it helps a little, but the nefarious code could still swap out the data capability, so you still need something like a reservation, it seems.

Reply:
I see you've added this. Looks good. I wonder about sealing on the load path though. Should be fine, but seems slightly messier than just saying whoever installs the code cap had to seal it. I guess that requires yet another new instruction though, or messing with csealentry to allow sealing with the new type as well